wasm/client.go

```diff
@@ -109,6 +109,89 @@ func Client_SendText(_ js.Value, args []js.Value) interface{} {
 	})
 }
 
+type FileWrapper struct {
+	file  js.Value
+	Size  int64
+	index int64
+}
+
+func NewFileWrapper(file js.Value) *FileWrapper {
+	size := file.Get("size").Int()
+	return &FileWrapper{file: file, Size: int64(size), index: 0}
+}
+
+func (fileWrapper *FileWrapper) Read(p []byte) (n int, err error) {
+	if fileWrapper.index >= fileWrapper.Size {
+		return 0, io.EOF
+	}
+
+	var uint8Array = js.Global().Get("Uint8Array")
+
+	// use Blob.slice(start, end) to read a part of the file.
+	start := fileWrapper.index
+	end := start + int64(len(p))
+
+	var (
+		bCh   = make(chan struct{}, 1)
+		errCh = make(chan error, 1)
+	)
+
+	fileSlice := fileWrapper.file.Call("slice", start, end)
```

Comment:
What happens if `fileWrapper.file` is undefined? I would expect JavaScript to say something like `cannot call "slice" on undefined` but `js.Value#Call` doesn't return an error. Perhaps it would be prudent to ensure that `fileWrapper.file` is at least not undefined, if not of the correct type (i.e. `js.Global().Get("File")`). It seems like it may make sense to do this in the constructor because otherwise we would have an invalid struct but that would require changing the constructor signature.

Reply:
Thanks.